src/main/java/net/hydromatic/sqllogictest/executors/JdbcExecutor.java

```diff
@@ -448,7 +448,8 @@ public void closeConnection() throws SQLException {
         boolean stop;
         try {
           stop = this.query(query, result);
-        } catch (SQLException ex) {
+        } catch (Throwable ex) {
```

Comment:
Catching `Throwable` is too broad and risky. Maybe we should be more conservative and catch `AssertionError` instead. I don't feel strongly about it do as you feel.

Reply:
If I was sure that this is the only exception thrown by Calcite (or, more generally, the executor under test) on error it would work, but I bet this is not the only case.

Reply:
In general we shouldn't try to recover from any kind of error but its not the first time I see `catch (Throwable)` so I am fine leaving it as such.

Reply:
In general I agree, but without catching Throwables all uncaught exceptions in the executor will cause immediate termination of the test program, which is not what we want. We want to run all the tests and make a summary of the errors.

Reply:
Every `Throwable` is either an `Error`, a `RuntimeException` or an `Exception`. `Exception`s are checked, so cannot happen by surprise. We agree that `RuntimeException` should be caught. But `Error`s are - by definition - not generally recoverable. For example, you should not be catching an `OutOfMemoryError`.

I would recommend keeping the catch clause for `SQLException`, and adding a catch clause for `RuntimeException`, but let `Error` fall through.